src/core/symbology-ng/qgssymbolv2.h

```diff
@@ -280,10 +281,26 @@ class CORE_EXPORT QgsMarkerSymbolV2 : public QgsSymbolV2
     QgsMarkerSymbolV2( QgsSymbolLayerV2List layers = QgsSymbolLayerV2List() );
 
     void setAngle( double angle );
-    double angle();
+    double angle() const;
+
+    /** Set "en masse" expression for angle
+     */
+    void setAngleExpression( const QString & exprStr );
+    /** Get "en masse" expression for angle
+     * @return empty if not an expession at the marker level
+     */
+    QString angleExpression() const;
 
     void setSize( double size );
-    double size();
+    double size() const;
+
+    /** Set "en masse" expression for size
+     */
+    void setSizeExpression( const QString & exprStr );
```

Comment:
@vmora for new methods like the en masse size these should definitely be stored as QgsDataDefined. There's two main reasons for this - one is that the behaviour of QgsDataDefinedButtons should be consistent between symbology and labeling/composer, and this requires storing the extra information like active/deactive which requires QgsDataDefined. The more important reason though is that it future proofs this work. I'm currently working on enhancements to QgsDataDefined to allow a system of inheritable properties which are evaluated against expression contexts (you can see some early discussion here https://github.com/qgis/QGIS-Enhancement-Proposals/pull/12 and an early version of the expression context work here https://github.com/nyalldawson/QGIS/tree/exp_context8 and here https://github.com/nyalldawson/QGIS/commit/cb5f244fc92c60418a25f8fa45fae4b95fb717c8 ). This work will make implementing things like this PR much easier, but unfortunately it's still a while off. In the meantime all new methods should utilise QgsDataDefined so they'll be ready to flip across.

I'm more than happy to help with this -- if it's not done now it will cause me a lot of work in future ;)

(btw, I started work on porting symbology to use QgsDataDefined some time ago, but haven't had a chance to finish this - the work in progress is here https://github.com/nyalldawson/QGIS/commit/6126e09418b49ec8a2258a43a85849fb80148c0b ).